### DISCUSSION TITLE
Is posix shell or BaSH required to use setup.py?

### DISCUSSION THREAD

**Discussion by mikofski** [NatLabRockies/solar-data-tools#32] [Q&A]
I am trying to use duramat/pvpro, and it says in the readme I should have the latest dev branches from solar-data-tools. So I tried to `pip install -e ../solar-data-tools` from my clone and it crashes on [`setup.py:L27`](https://github.com/slacgismo/solar-data-tools/blob/master/setup.py#L27) presumably b/c I am in a Windows CMD terminal, and so I don't have the `tail` shell command.

**Comment by mikofski** [comment]
also, I think [versioneer](https://pypi.org/project/versioneer/) does what you want to do

**Reply by mikofski** [reply]
or better yet I think what you want is `git describe --long`:
https://stackoverflow.com/questions/35522047/pep440-compliant-and-git-describe-info-available-from-deployed-package
```
$ git describe --long
v0.4.3-6-gda72e35
```

**Comment by bmeyers** [comment]
Hi Mark! That advice on installing the `dev` branch is already stale. Now, PV-Pro (any all other projects) just depends on the current published version, which is v0.4.5 as of this afternoon. So, hopefully your issue is resolved, but please reach out if you have any other problems!

**Reply by bmeyers** [reply]
By the way, I just put together this repository to try to help people get up and running faster: https://github.com/slacgismo/pvinsight-onboarding

Please let me know if this is useful or if you have any suggestions!

**Reply by mikofski** [reply]
Hi @bmeyers,

The issue is still there in the `setup.py` for `solar-data-tools`
https://github.com/slacgismo/solar-data-tools/blob/a22f319ea619e41fb6da20105b30f58bcadb39dd/setup.py#L27

The problem is that Windows does not have the `tail` command so you cannot run `pip install -e path/to/solar-data-tools`. The error is:

    FileNotFoundError: [WinError 2] The system cannot find the file specified

I suppose this is only an issue for "developers" who would probably be savvy enough to figure it out on their own, however, it requires them to either use a posix environment like WSL, msys, or git-bash or make changes to the source to install in a `pvi-dev` conda env on a Windows machine.

My recommendation would be either:
1. change this command to `git describe --long`
    ```python
    # get the most recent git tags from the cmd line
    latest_git_tag = subprocess.Popen(['git', 'describe', '--long'], stdout=subprocess.PIPE, stderr=subprocess.STDOUT)
    latest_version = latest_git_tag.communicate()[0].strip()
    # b'v0.4.3-34-ga22f319'
    ```

1. or better yet, just use [versioneer](https://pypi.org/project/versioneer/). Many projects use versioneer even though these commands seem so simple, but using the same tool standardizes the output. For an example see either [pvlib](https://github.com/pvlib/pvlib-python/blob/933d0d80408651801de9755e0e427f4abb02fd6d/setup.cfg#L6-L12) or [rdtools](https://github.com/NREL/rdtools/blob/3944c62793315a5be7776c6bbb23bacb341a2f23/setup.cfg#L5-L11)

Regarding the new [pvinsight-onboarding](https://github.com/slacgismo/pvinsight-onboarding) repo, it looks great. If I have any suggestions, I'll start up a discussion over there. Thanks so much for working on these tools and for putting in extra time to get others up and running. It really makes a difference, and I appreciate your effort!

**Reply by bmeyers** [reply]
Thank you Mark! We are starting a more concerted effect with additional staffing to work on software development and deployment, expecting to start full effort around May 1, so this is super helpful.
